src/js/modules/articles/actions.js

```diff
@@ -1,11 +1,67 @@
 import axios from 'axios';
 import * as t from './actionTypes';
+import { getSections } from '../sections/selectors';
 
-const apiUrl = 'http://api.icndb.com/jokes/random';
+const apiUrl = 'https://api.stuyspec.xyz/articles';
 
+//TODO: need a way to call the fetch function
 export const fetch = () => {
-	return {
-		type: t.FETCH_ARTICLE,
-		payload: axios.get(apiUrl),
-	};
+  return (dispatch, getState) => {
+    dispatch({ type: t.FETCH_ARTICLE_PENDING });
+    //axios.get(apiUrl, {'headers': {'X-Key-Inflection': 'camel'}}) TODO: wait until Nic uses olive branch
+    axios.get(apiUrl)
+      .then((info) => {
+        // This current code will dispatch the FETCH_ARTICLE_REJECtED because the API still has null values
+        if (articleValidator(info.data)) {
+          const sections = getSections(getState());
+          dispatch({
+            type: t.FETCH_ARTICLE_FULFILLED,
+            payload: info.data,
+            sections: sections,
+          })
+        } else {
+          dispatch({ type: t.FETCH_ARTICLE_REJECTED, payload: 'Invalid article array' })
+        }
+      })
+      .catch((err) => {
+        dispatch({ type: t.FETCH_ARTICLE_REJECTED, payload: err })
+      })
+  };
 };

```

Comment:
(1) `articleValidator` is a function that returns a boolean. That usually means the function is an adjective. Conventionally, you would name the function something like `articleIsValid()`

(2) `info` is not a good name for the response you receive from an API. It should be called `response`.

(3) REMOVE the `sections` field when you dispatch the action with type `FETCH_ARTICLE_FULFILLED`. Since every single action has that same `sections` field, why not import the sections selector and the sections in `modules/articles/reducers.js`, where you actually use the sections?